components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/RecommendationsApiServiceImpl.java

```diff
@@ -54,7 +54,7 @@ public Response recommendationsGet(MessageContext messageContext) {
             String requestedTenantDomain = apiConsumer.getRequestedTenant();
 
             if (apiConsumer.isRecommendationEnabled(requestedTenantDomain) &&
-                    userName != APIConstants.WSO2_ANONYMOUS_USER) {
+                    !userName.equals(APIConstants.WSO2_ANONYMOUS_USER)) {
```

Comment:
let's use the format "KNOWN_STRING".equals(derived_string) so that there is no chance for a null pointer

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/8895